src/lib/database/store.ts

```diff
@@ -665,28 +738,122 @@ const initializer = immer<Database>((set, get) => ({
+      phone_sdk_installations: [
+        ...get().phone_sdk_installations,
+        new_installation,
+      ],
     })
-    return installation
+    return new_installation
   },
 
   getPhoneSdkInstallation(params) {
+    const client_session = get().client_sessions.find(
+      (cs) => cs.client_session_id === params.client_session_id,
+    )
+
+    if (client_session === undefined) {
+      return
+    }
+
     return get().phone_sdk_installations.find(
       (installation) =>
         installation.workspace_id === params.workspace_id &&
-        installation.client_session_id === params.client_session_id &&
+        installation.user_identity_id === client_session.user_identity_id &&
         installation.ext_sdk_installation_id === params.ext_sdk_installation_id,
     )
   },
 
   addInvitation(params) {
-    const invitation = phone_invitation.parse(params)
+    const client_session = get().client_sessions.find(
+      (cs) => cs.client_session_id === params.client_session_id,
+    )
+
+    if (client_session?.user_identity_id === undefined) {
+      throw new Error(
+        "Could not find client session associated with a user identity!",
+      )
+    }
+
+    const invitation_id = get()._getNextId("invitation")
+    const new_invitation: PhoneInvitation = {
+      invitation_id,
+      invitation_type: params.invitation_type,
+      invitation_code: params.invitation_code,
+      phone_sdk_installation_id: params.phone_sdk_installation_id,
+      workspace_id: params.workspace_id,
+      user_identity_id: client_session.user_identity_id,
+
+      assa_abloy_credential_service_id: params.assa_abloy_credential_service_id,
+    }
+
     set({
-      phone_invitations: [...get().phone_invitations, invitation],
+      phone_invitations: [...get().phone_invitations, new_invitation],
     })
-    return invitation
+    return new_invitation
+  },
+
+  assignInvitationCode(params) {
+    const invitation = get().phone_invitations.find(
+      (invitation) => invitation.invitation_id === params.invitation_id,
+    )
+
+    if (invitation === undefined) {
+      throw new Error("Could not find invitation!")
+    }
+
+    const updated_invitation = {
+      ...invitation,
+      invitation_code: `${invitation.invitation_id}:code`,
+    }
+
+    set({
+      phone_invitations: [...get().phone_invitations, updated_invitation],
+    })
+
+    return updated_invitation
+  },
+
+  getEnrollmentAutomations(params) {
+    const client_session = get().client_sessions.find(
+      (cs) => cs.client_session_id === params.client_session_id,
+    )
+
+    if (client_session?.user_identity_id === undefined) {
+      throw new Error(
+        "Could not find client session associated with a user identity!",
+      )
+    }
+
+    return get().enrollment_automations.filter(
+      (automation) =>
+        automation.user_identity_id === client_session.user_identity_id,
+    )
   },
 
   getInvitation(params) {
```

Comment:
do we still need this method now?

Reply:
used by the `internal/phone/get_invitation` endpoint I think

Reply:
ill prob clean that up in a subsequent pr, probably better to keep it a little dry